aten/src/ATen/native/cpu/DistanceOpsKernel.cpp

```diff
@@ -197,39 +187,18 @@ struct PDist {
     // The only way to parallelize and avoid locking requires parallelizing
     // over the columns of the input, i.e. we compute the gradient for the
     // first section of each vector independentaly of the second section, etc.
-    int64_t mv = (m + Vec::size() - 1) / Vec::size(); // number of Vecs in a row rounded up
-    at::parallel_for(0, b * mv, internal::GRAIN_SIZE / (8 * n * n), [=, &pvec](int64_t start, int64_t end) {
-      const int64_t l = start / mv;
-      const int64_t v = start % mv;
-
-      const scalar_t * self_l = self_start + l * n * m;
-      const scalar_t * self_v = self_l + v * Vec::size();
-
-      const scalar_t * dist_l = dist_start + l * combs;
-      const scalar_t * grad_l = grad_start + l * combs;
-
-      scalar_t * res_l = res_start + l * n * m;
-      scalar_t * res_v = res_l + v * Vec::size();
-
-      while (start != end) {
-        backward_down_column<F>(self_v, res_v, grad_l, dist_l, pvec, n, m, std::min(int(m - (self_v - self_l)), Vec::size()));
+    at::parallel_for(0, m / Vec::size(), internal::GRAIN_SIZE / (8 * n * n), [=, &pvec](int64_t l, int64_t end) {
```

Comment:
I'd try to very carefully audit exactly what variables you're capturing here, since it can make it harder for the compiler to inline the code and avoid costly function calls.. Ideally you catch none. pvec is probably better initialized within that function.

Reply:
Came from the revert.